### PR TITLE
Add hardware profile table

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: https://issues.redhat.com/browse/RHOAIENG-16254

## Description

Added the table for the hardware profiles.

Main table:
Screenshot 2024-12-12 at 3 59 24 PM

Expandable sections:
There are 3 tables inside, if there is no data in the table, the corresponding nested table will be hidden.
Screenshot 2024-12-12 at 4 00 13 PM

Filters:
You can filter by Name and Enable columns.
Screenshot 2024-12-12 at 4 01 27 PM
Screenshot 2024-12-12 at 4 02 05 PM

Disable hardware profile modal:
Screenshot 2024-12-12 at 4 03 00 PM

Delete modal:
Screenshot 2024-12-12 at 4 03 20 PM

## How Has This Been Tested?

1. Create some hardware profiles CRs on the clusters in the `opendatahub` namespace
2. Try to set the empty array to `nodeSelectors`, `identifiers` and `tolerations` fields in `spec` to see how the expandable section shows the tables
3. Try to toggle the enable switch to see the modal and how it performs
4. Try the delete function
5. Try the filters, make sure it can filter by the name and enable status

## Test Impact

Added cypress tests for this page.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by DaoDaoNoCode** [comment]
@vconzola Hi, can you check the screenshots?

File: frontend/src/pages/hardwareProfiles/const.ts
Comment (DaoDaoNoCode) [inline frontend/src/pages/hardwareProfiles/const.ts:1]: @dpanshug I prefer to keep everything in the root `const.ts` file because we use different names for the column variables. Just let you know in advance so it's easier for you to rebase if needed.

**Comment by vconzola** [comment]
@DaoDaoNoCode A couple comments questions...

1. What font size are you using for the section headers (Node resources, Node selectors, Tolerations)? They look kinda big.
2. We should not require confirmation when disabling a profile. We don't confirm when disabling other resources, like notebook images and serving runtimes.
3. Let's change the column header to "Enabled" instead of "Enable".
4. All of the content will need a review by @kaedward. Do you want to do that now or after the code is merged?

**Comment by DaoDaoNoCode** [comment]
@vconzola Hi, thanks for the feedback.
1. I am using `h6` variant for `Title` component, which I believe size is `md` here. This is the smallest size of the `Title` component, I can change it to normal Text and add bold style manually if you think it's too big now.
2. I can remove this disabled modal. I saw it when disabling the accelerator profiles so I just copied what they do
Screenshot 2024-12-16 at 9 20 15 AM
3. Will do
4. I prefer to get the content reviewed after all the codes are merged

**Comment by DaoDaoNoCode** [comment]
@vconzola I addressed the feedback as suggested. Can you check it again?

Screenshot 2024-12-16 at 9 49 59 AM

**Comment by vconzola** [comment]
Thanks @DaoDaoNoCode This LGTM. I think it looks better with the smaller font for the section headers. Regarding the confirmation dialog, leave it out for now. I've started a thread within the UXD channel to talk about this, because we're inconsistent in different parts of the UI currently. I'll let you know what we decide, but I don't think that should block you from getting your PR merged, right?

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

looks great

**Pull request merged**
